contracts/Removal.sol

```diff
@@ -331,6 +332,38 @@ contract Removal is
     }
   }
 
+  /**
+   * @dev See {IERC1155-safeTransferFrom}.
+   */
+  function safeTransferFrom(
+    address from,
+    address to,
+    uint256 id,
+    uint256 amount,
+    bytes memory data
+  ) public override whenNotPaused {
+    if (_msgSender() != address(_market)) {
```

Comment:
the only reason we have to keep these transfer functions around (`safeTransferFrom` and `safeBatchTransferFrom`) is because the Market contract uses them while completing purchases (to send to Certificate) and in the `withdraw` implementation.  We already don't allow anyone to own a removal besides the original supplier, the market, the certificate, or the 0 address. By adding this check that only the market can use this function, we prevent a supplier from using this function directly to transfer their tokens to either the market or certificate contracts (or even 0 address)